src/components/HappinessNumber.tsx

```diff
@@ -0,0 +1,99 @@
+import ArrowDownIcon from "../assets/ArrowDownIcon";
+import ArrowUpIcon from "../assets/ArrowUpIcon";
+import Column from "./layout/Column";
+import { useEffect, useRef, useState } from "react";
+
+/**
+ * Component represeting the Happiness number.
+ * Has editable and fixed states.
+ * @param value the value of the happiness number
+ * @param onChangeValue what to do with the value after it has been edited and validated
+ * @param editable whether the Happiness number is in editable state
+ * @returns
+ */
+export default function HappinessNumber({
+  value,
+  onChangeValue,
+  editable,
+}: {
+  value: number;
+  onChangeValue: (n: number) => void;
+  editable: boolean;
+}) {
+  const [currentHappiness, setCurrentHappiness] = useState(value);
+
+  const updateHappinessTimeout = useRef<number | undefined>(undefined);
+  /**
+   * Update Happiness function
+   * Function responsible for validating and changing the "true" happiness value
+   * Where the true happiness value is the happiness value that is final after
+   * the happiness validator validates the input
+   *
+   * Precondition: `currentHappiness` is in the range from 0 to 10 inclusive
+   */
+  const updateHappiness = () => {
+    const validHappiness = Math.round(currentHappiness * 2) / 2;
+    onChangeValue(validHappiness);
+    setCurrentHappiness(validHappiness);
+  };
+
+  useEffect(() => {
```

Comment:
don't want to be too picky but can you add some docs about how you are making sure the happiness values are legal?